03_cross_cancer_classification/plot_holdout_pca.ipynb

```diff
@@ -0,0 +1,385 @@
qG7dQLVtG6D4+MpZxfSnn47qxx8PqW3bewrdJzk5WStXvquoqKYKCwt3v79x43o98MBDuv32OxQeHq7PPttephjateugb77ZI0lKS0vT8ePH9e23+yVl9yw7ePCA7rmnQ6E/v3TpmzpxIk7r1m3UkiXL9M03Xxe6b3T0ZN1xx52aMCFazzwzUT179tann14d4rljx3bdfnsL1amTPZT1k08+1pw5f9amTVuVmZmhV16ZK0n67LPteu+9d/TnPy/Upk1bdMMN/6EZM6aW6fwrgz3thEK+vEthW0IV8uVdsqedqLRjW+Eay/HZZ59q+fIYrVixSv/4xwatXPmu3nlnlRYv/pvWrPm7jh+PK/Lnu3a9V489NkodO3bW8uUxSk9P19SpkzV8+Eht2bJDq1ev1Y8/HtK2bVvdP7N9+1bNm7dA//jHZhmGoffeW65atWpp4cLXFRoaps8/36XQ0LBynRcAAPAeDDEGAACFGjKkho4ds8nlsunYsezXX32V7pFjjRnzX7LbbcrMzNKVK5fVps09iopq5t5+6VKSunXrKJfLpczMTAUGBqlTpy76y1/+6t7n3/8+qOTkZLVr116S9PDD/bV27Yfq2bP3NZ+T24svvqw2bfImiu68s6WSkpJ0+vQpnTwZr7vuuls//viD4uNP6NSpnxUZGaVatWoVej6ffrpdkyZNcSeWhg8fqcWL3yhRWXTteq/mz39Z8fEn1LhxhLZt26r77+/j3j5gwCA1bZpdNmPHjtfo0SOUkZGhTZs2aPDgoWrSJFKS9OSTT+veezvo//7vpBo2bFSiY1em4H0DZE/9STa5ZE/9ScH7BuhSx8J7WpaX1a6xHA899LBCQkIUEhKi2rXrqE+fh1SzZk3VrFlTderU0S+/nHX/TkvC399f7767SvXrN1BKSrISEhIUFhamhITz7n169uylG274D0lSp05dtHPnlyX+fAAAUPWQIAQAAIWKjc1ODkqSy2VTbGwxP1AOb7/9jnt+uF9/vaAXX3xBM2dO0/z5r0qSQkJC3fO67d+/TzNmTNGtt96WZ4GIDRs+UlJSovr06SlJcjqdSkpK1JEjP6p585uv+Zyi+PlVU+vW/6m9e/fo5MmTatXqbvn7++vbb/crLu6Y2rfvWOTP//rrBV133dXY6tW7wf3vZ599SgcPfidJuuOOO/Xqq3/N87PBwTXVrl0HffrpNj344MM6fPiQ5s//i3t7/foN3P++7rrrlJmZqUuXkvTLL79oyZI3tWzZUvd2m82ms2fPWjJBaE87JptckpSdJEw75tHjWe0ayxESEuL+t91uV82aNd2vbTabXC5Xqc7T4XBo584vtXr1KklSVFQzpaen5/mc3D0i/fz85HKVf/5CAADgvUgQAgCAQkVFGTp2LDs5aLcbioqqnCRC7dp11K/fAP3P/xQ8PLZVq7s0bdpMTZv2J9Wv30AtW7ZSSkqyduzYrtdfX6z69eu791248BX9/e8f6rnnXih1HO3atddXX+3WqVM/a+bMF+Tv76+9e/foyJHDevnl+UX+bJ06dXX27Fl30ih37638CcGC9OjRS2+/vVghIaFq166DgoKC3dsuXEhw//uXX86qevUaCg0NU506dTRs2HD16dPXvf3EieN5EopW4gps6u5BaMguV2DTSju2Va4xKTsJWBJ2uz3PfIe5557M7d//Pqhly5Zq2bIYNWzYUJI0fvzjZYoNAAD4BuYgBAAAhfrgg8tq2tSQw2GoaVNDH3xwuVKOm5ycrE2bNuq2224vdJ+OHTurR49eevHFWUpPT9cnn3ys+vUb6I47Wqh27Tru/x58sK8+/XSrEhMvljqOtm3ba+/eb5SQkKDIyCjdddfd2rXrX8rKynIP8S1Mr1736513/qaEhAT9+usFvffeiiL39/evptTUVPfrdu06KCHhvP7xj/Xq0aNXnn3Xrv27Tp8+pZSUZC1e/Ka6d++hatWqqXfvPnr//ZX6+ef/k8vl0t//vlr//d8jlZ7umWHh5ZVy1xq5gprJsDnkCmqmlLvWVNqxrXKNlUaDBg21bdsWZWVlas+er/XDD/92b/P393dfP6mpKbLbHapRo7qcTqc2b/6nDhz4rkSrE/v7V1NGxpVyLbwCAAC8Dz0IAQBAoRo3Njw252B+o0aNkN1uk81mk59fNd19d2vNmjW7yJ955pmJGjKkvxYv/qv279+n7t17XrPP3Xf/p0JDw7Rx43rdemvhyaCC1K5dW/Xr19f11/9BNptN9es3UFhYWJGLk+QYPXqMUlNTNXRof9WoEaAePXrlWXgkv+7de2nBgnk6e/aMpk2bqWrVqqlLl3v1+eefqW3bdnn2vfXW2zR5crTOnz+nTp26KDo6exXlXr3u16VLSYqOflq//fabGjdurAULFuUZwmolrsAIj845mJ8Vr7HSmDRpihYtWqD169fqrrta54mlXbsOWrNmtQYOfFirV69T167dNGzYQNntDt10003q3buP4uOLXwQmKqqZIiIi1aNHF7377vtq0KChx84HAABYh80wDCYcAQCgijp06EfdcIP15p5Dybzzzt90/vw5TZnyP+73xo0bo65d79WAAYNNjAy+6syZk7rllpvNDgMAAFQwhhgDAABYzMWLF3Xo0A/asOGjPPMJAgAAAJ5AghAAAMBiDhz4VuPHP65u3brr5ptvMTscAAAAVHEMMQYAoApjiDGAisQQYwAAqiZ6EAIAAAAAAAA+jAQhAABVHIMFAFQE6hIAAKouEoQAAFRhDodDTmeW2WEAqAKcziz5+TnMDgMAAHgACUIAAKqwWrXCdenSRRmGy+xQAHgxw3ApKemiwsPDzQ4FAAB4AIuUAABQhblcLv38889KTU0Vf/EBlJXNJgUFBalBgway2+ljAABAVUOCEAAAAAAAAPBhPP4DAAAAAAAAfBgJQgAAAAAAAMCHkSAEAAAAAAAAfBgJQgAAAAAAAMCH/X/Y8ebG8ytqyAAAAABJRU5ErkJggg==\n",
```

Comment:
Neat! So these BRAF scatter plots look as I would expect given the flatness of the curve. However I would've expected the separability to be more muddled in EGFR which actually drops in performance with more data held out (slide 9). Having trouble figuring out how to explain this trend. Maybe this means that the data is nonlinearly separable in some dimension (since the mutated ones are clustering together) but as you hold out more and more you are loosing the signal from the mutated ones. So there seems to be a class imbalance issue here.

Reply:
Yeah, I think there are a few things to note (but I'm guessing a bit at this point):

* Maybe the relationship between features and labels is complex/multi-dimensional, and not really clearly visible in the 2D UMAP plots (but the model can find it). EGFR in lung cancer is a particularly interesting case since it's a pretty well-known biomarker that correlates with response to certain targeted therapies (see, e.g. [this paper](https://doi.org/10.1126/science.1099314)), so I expected there to be some relationship between EGFR mutation status and the gene expression data, but it's not really as strong as I expected across holdout sets.
* Maybe the label imbalance is driving the decrease in performance, like you're suggesting (even though there's still a decent amount of training data overall in the 95% case, there are very few positive labels so it may be hard to train a good model)
* Maybe we're just sampling bad training sets (but I doubt it, results across 3 random seeds were pretty similar)

If I had to guess I'd say what we're seeing represents a combination of all of the above, but I'm not sure!